src/LayerEditor/ui/data/le_model.py

```diff
@@ -266,11 +284,20 @@ def save(self):
         geo_model.interfaces = interfaces_data
         geo_model.layers = self.blocks_model.save()
 
+        last_node_set = self.decompositions_model.decomps.index(self.gui_block_selector.value.decomposition)
+
+        if self.gui_surface_selector.value in self.surfaces_model.surfaces:
```

Comment:
Why should it not? Selector.value is a value and Im asking if that value is contained in a list. This is done to prevent list.index() throwing ValueError.